### PR TITLE
2.5.0 Magento Plugin Update Add New Endpoints

### PR DESCRIPTION
**Pull request by justin-robertson-git**
# Magento Plugin Update Add New Endpoints

This version adds new endpoints following the [ShipEngine Connect Order Source Spec](https://connect.shipengine.com/orders/)

It implements the following endpoints (some not currently documented but coming soon)
- SalesOrdersExport
- ShipmentNotification
- InventoryPush
- InventoryFetch
- Diagnostic Routes

Security is handled with the use of the BaseAuthorizedController which expects an Bearer Token with the users provided ShipStation Token.

### CODE REVIEW

**Comment by danpoage** [comment]
86 files of PHP code, I think we know who passed Ben's test, doctor...

File: Api/Model/OrderSourceAPI/Models/RequestedFulfillmentExtensions.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<?php
+namespace Auctane\Api\Model\OrderSourceAPI\Models;
+
+class RequestedFulfillmentExtensions
+{
+    /** @var string|mixed|null Maps to custom field 1 in Shipstation */
+    public ?string $custom_field_1;
```
Comment (danpoage) [inline Api/Model/OrderSourceAPI/Models/RequestedFulfillmentExtensions.php:7]: These fields names look at home, written in this language.

**Comment by danpoage** [comment]
This code all looks reasonable, as far as PHP goes. What is the plan for testing this? Are we going to be able to do so in a low-risk way, or is this something we will deploy to an immediate chorus of complaints?

**Comment by justin-robertson-git** [comment]
> This code all looks reasonable, as far as PHP goes. What is the plan for testing this? Are we going to be able to do so in a low-risk way, or is this something we will deploy to an immediate chorus of complaints?

So I was able to test / develop by having a local version of Magento 2.5.7 running on my machine then I just git cloned this repo into the appropriate directory. All endpoints were tested using postman.

**Comment by justin-robertson-git** [comment]
The main drive for this will be the inventory functionality, before we publish the official version through Adobe, we will probably be sending pre release copies to test users as I build out the connector module.

**Review by danpoage — APPROVED** [review]

**Pull request merged**